code/game/objects/obj_defense.dm

```diff
@@ -11,6 +11,9 @@
 	damage_amount = run_obj_armor(damage_amount, damage_type, damage_flag, attack_dir, armour_penetration)
 	if(damage_amount < DAMAGE_PRECISION)
 		return
+	if(SEND_SIGNAL(src, COMSIG_OBJ_TAKE_DAMAGE, damage_amount, damage_type, damage_flag, sound_effect, attack_dir, armour_penetration) & COMPONENT_NO_TAKE_DAMAGE)
+		return
+
```

Comment:
Why put this here?

Reply:
Because this is after all of the checks and damage mitigation. At this point the object is guaranteed to actually take damage.